static/app/views/automations/components/automationListTable/conenctedDetectors.tsx

```diff
@@ -0,0 +1,127 @@
+import {ClassNames} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {Hovercard} from 'sentry/components/hovercard';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import LoadingError from 'sentry/components/loadingError';
+import Placeholder from 'sentry/components/placeholder';
+import {EmptyCell} from 'sentry/components/workflowEngine/gridCell/emptyCell';
+import {tn} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {DetectorLink} from 'sentry/views/detectors/components/detectorLink';
+import {useDetectorsQuery} from 'sentry/views/detectors/hooks';
+
+type AutomationListConnectedDetectorsProps = {
+  detectorIds: string[];
+};
+
+const MAX_DISPLAYED_DETECTORS = 5;
+
+function ConnectedDetectorsBody({detectorIds}: {detectorIds: string[]}) {
+  const shownDetectors = detectorIds.slice(0, MAX_DISPLAYED_DETECTORS);
+  const {data, isPending, isError} = useDetectorsQuery({
+    ids: detectorIds.slice(0, MAX_DISPLAYED_DETECTORS),
+  });
+  const hasMore = detectorIds.length > MAX_DISPLAYED_DETECTORS;
+
+  if (isError) {
+    return <LoadingError />;
+  }
+
+  if (isPending) {
+    return (
+      <div>
+        {Array.from({length: shownDetectors.length}).map((_, index) => (
+          <HovercardSkeletonRow key={index}>
+            <Placeholder height="20px" width="100%" />
+            <Placeholder height="18px" width="70%" />
+          </HovercardSkeletonRow>
+        ))}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      {data?.map(detector => {
+        return (
+          <HovercardRow key={detector.id}>
+            <InteractionStateLayer />
+            <DetectorLink
+              detectorId={detector.id}
+              name={detector.name}
+              createdBy={detector.createdBy}
+              projectId={detector.projectId}
+              disabled={detector.disabled}
+            />
+          </HovercardRow>
+        );
+      })}
+      {hasMore && (
+        <MoreText>
+          {tn('%s more', '%s more', detectorIds.length - MAX_DISPLAYED_DETECTORS)}
+        </MoreText>
+      )}
+    </div>
+  );
+}
+
+export function AutomationListConnectedDetectors({
+  detectorIds,
+}: AutomationListConnectedDetectorsProps) {
+  if (!detectorIds.length) {
+    return <EmptyCell />;
+  }
+
+  return (
+    <ConnectedDetectors>
+      <ClassNames>
+        {({css}) => (
```

Comment:
what does this do? 😮 i also see `bodyClassName` being passed to the Hovercard, i'm just not familiar with the component

Reply:
This is almost never necessary, but we need it when generating a css classname outside of a styled component. Since HoverCard requires you to send it as a string (instead of most comonents which accept a single `className` we have to do it this way